proto/buf/alpha/registry/v1alpha1/plugin_curation.proto

```diff
@@ -114,6 +116,49 @@ service PluginCurationService {
   rpc GetLatestCuratedPlugin(GetLatestCuratedPluginRequest) returns (GetLatestCuratedPluginResponse);
 }
 
+// PluginCodeGenerationService generates code using remote plugins.
+service PluginCodeGenerationService {
+  // GenerateCode generates code using the specified remote plugins.
+  rpc GenerateCode(GenerateCodeRequest) returns (GenerateCodeResponse);
+}
+
+message GenerateCodeRequest {
+  // The image to run plugins against to generate the desired file outputs.
+  //
+  // All image files that are not imports and not well-known types will be generated.
+  // If you want to filter what files are generated, modify the image.
+  // If you want to include imports, set include_imports.
+  buf.alpha.image.v1.Image image = 1;
+  repeated PluginGenerationRequest requests = 2;
+  // Include imports from the Image in generation.
+  bool include_imports = 3;
+  // Include Well-Known Types from the Image in generation.
+  //
+  // include_imports must be set if include_well_known_types is set.
+  bool include_well_known_types = 4;
+}
+
+message GenerateCodeResponse {
+  // Responses from each plugin execution.
+  // The order of each response matches the order in the GenerateCodeRequest.
+  repeated PluginGenerationResponse responses = 1;
+}
+
+// Request for performing code generation using the specified plugin.
+message PluginGenerationRequest {
+  // The plugin to execute.
+  CuratedPluginReference plugin = 1;
```

Comment:
Instead of the previous `PluginReference` type which combined remote/owner/name/version with options, split it out into two separate fields here.

Reply:
I quite prefer this instead of overloading the `PluginReference` which contains details about identity _and_ code generation options.

Reply:
It's a bit verbose, but we might want to expand the field name to `plugin_reference`. We do this kind of naming elsewhere, too. We can keep the `curated` prefix out of it though.

If we ever support user-facing plugins, we'll definitely need to get rid of all this `Curated` prefix stuff (but we've already discussed that elsewhere).

Reply:
Fixed.